assembly/src/lib.rs

```diff
@@ -54,26 +54,42 @@ const MAX_U32_ROTATE_VALUE: u8 = 31;
 /// The maximum number of bits allowed for the exponent parameter for exponentiation instructions.
 const MAX_EXP_BITS: u8 = 64;
 
+/// The maximum length of a procedure's name.
+const MAX_PROC_NAME_LEN: u8 = 100;
+
 // MODULE PROVIDER
 // ================================================================================================
 
 /// The module provider is now a simplified version of a module cache. It is expected to evolve to
-/// a general solution for the module lookup
+/// a general solution for the module lookup.
 pub trait ModuleProvider {
-    /// Fetch source contents provided a module path
-    fn get_source(&self, path: &str) -> Option<&str>;
-
     /// Fetch a module AST from its ID
     fn get_module(&self, id: &ProcedureId) -> Option<NamedModuleAst<'_>>;
 }
 
 // A default provider that won't resolve modules
 impl ModuleProvider for () {
-    fn get_source(&self, _path: &str) -> Option<&str> {
-        None
-    }
-
     fn get_module(&self, _id: &ProcedureId) -> Option<NamedModuleAst<'_>> {
         None
     }
 }
+
+// LIBRARY
+// ================================================================================================
+
+/// TODO: add docs
+pub trait Library {
```

Comment:
One alternative to provide more clarification is to rename `ModuleProvider` to `ParsedModuleProvider` and `Library` to `SourceModuleProvider`. They are more verbose, but they should be immediately clear regarding their roles

Reply:
I actually think we should refactor the overall module provide structure a bit. I'll write this up in an issue later.